src/components/editor/Controls.tsx

```diff
@@ -12,37 +13,70 @@ const Label = ({ text, children }: {
 	</label>
 }
 
-export default function Controls({ exportHandler, exporting }: {
-	exportHandler: (
-		data: {
-			fps: number,
-			size: number,
-			speed: number,
-			bounce: boolean,
-			loop_gif: boolean
-		}) => void,
-	exporting: boolean
+const getEstimatedFileSize = (
+	fps: number,
+	width: number,
+	height: number,
+	quality: number,
+	durationInFrames: number
+) => {
+	const durationInSeconds = durationInFrames / CONSTANTS.previewFps;
+	const totalFrames = durationInSeconds * fps;
+	const qFactor = quality / 100;
+
+	const totalPixels = (width * height * totalFrames) / 3;
+
+	const totalBytes = (totalPixels * qFactor + 1.5) / 2.5
```

Comment:
And by 2.5 here?
Also, shoudn't 1 pixel be 3 bytes ( one each for r, g, b )? so do we need to multiple by 3 somewhere?

Reply:
GPT explained this as "compression compensation" so I've let it be. Tested this on a couple of GIFs and this algo seems to get in the right ballpark, will come back to refine this once the rest of the things are sorted.

Reply:
Oh 0kay. Makes sense.